internal/report/write.go

```diff
@@ -0,0 +1,81 @@
+package report
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/rs/zerolog/log"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/util/retry"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	polReport "sigs.k8s.io/wg-policy-prototypes/policy-report/pkg/api/wgpolicyk8s.io/v1beta1"
+)
+
+// Write instantiates the passed namespaced PolicyReport if it doesn't exist, or
+// updated a new one if one is found
+func Write(report *polReport.PolicyReport) error {
```

Comment:
nitpicking, maybe this should be named `Save` ?

Reply:
Yes, better. Renamed (now it is at internal/report/store.go).